app/routers/action.py

```diff
@@ -0,0 +1,364 @@
+################################################################################
+# Copyright (c) 2025 Hackerbot Industries LLC
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# Created By: Allen Chien
+# Created:    April 2025
+# Updated:    2025.05.16
+#
+# This script contains the action Fast API endpoints.
+#
+# Special thanks to the following for their code contributions to this codebase:
+# Allen Chien - https://github.com/AllenChienXXX
+################################################################################
+
+
+from fastapi import APIRouter, Request, HTTPException, Depends
+from app.utils.request_helpers import get_required_param
+from fastapi.responses import JSONResponse
+from app.dependencies import get_robot 
+
+router = APIRouter()
+
+@router.post("/core")
+async def core_post(request: Request, robot = Depends(get_robot)):
+    """
+    Handle POST requests to the /core endpoint.
+
+    This function processes incoming requests to perform core operations on the robot.
+    It supports the following methods:
+    - 'ping': Checks the connectivity with the robot core.
+    - 'settings': Configures robot settings, including enabling/disabling JSON responses
+      and time-of-flight sensors.
+
+    Args:
+        request (Request): The incoming HTTP request containing JSON data with the method
```

Comment:
Rather than using just Request, I think it might be better to create types based on the particular inputs you're expecting to get. This would work both in the function definition and in the docstring which I think would help people understand what each API endpoint should be receiving.